doc/developer/design/20231027_refresh_every_mv.md

```diff
@@ -0,0 +1,164 @@
+
+Contrast the above behavior of upper to what _A._ alone would get us: With only _A._, the upper would be mostly stuck near the _beginning_ of the current refresh interval. With _B._, the upper will be mostly standing at the _end_ of the current refresh interval. This behavior of upper gets us halfway to achieving success criterion 3.: The materialized view's upper won't impede timestamp selection of queries that also involve other objects, because the MV's upper will be far ahead of the sinces of other, normal objects (that don't have a refresh interval specified).
+
+We still need to consider how the since frontier of the materialized view moves. Crucially, it shouldn't jump forward to near the time of the next refresh, because that would again preclude timestamp selection when querying the MV together with other objects, violating success criterion 3., because the since of the MV would be far ahead of the uppers of normal objects. When initially designing this feature, we mistakenly thought that without some further code modifications the since would jump forward to the next refresh because that is the since that the Compute controller would request on the MV: tailing the upper by 1 second (or more generally, by the compaction window of the MV). However, it turns out that the Adapter is keeping the sinces of all objects from jumping into the future in any case: The Coordinator is keeping read holds on all objects, and ticks these read holds forward every second. ([See `Coordinator::advance_timelines`](https://github.com/MaterializeInc/materialize/blob/b89d4961f8a8ccd172be2dc46f7b83bb00b9b871/src/adapter/src/coord/timeline.rs#L645).)
+
+A code modularity argument could be made for not relying on the Adapter to hold the sinces near the current time: One could say that the Compute Controller should be keeping sinces reasonable without relying on other components (the Adapter). However, I would counter this argument by saying that it is the Adapter's responsibility to keep sinces from moving into the future, because it's the Adapter that will be issuing the queries that need the sinces to be near the current time. Therefore, I'd say it's ok to rely on Adapter holding back the sinces from jumping far into the future.
```

Comment:
One thing that seems reasonable, at least to record and ponder, is that someone understands the *granularity* of update, and e.g. if "daily" there is no reason to maintain the data more compact than the write frontier minus one day. There are no changes *within* the day, and no reason to move the trailing compaction frontier forward as it will accomplish nothing until it can tick forward an entire day as well.

If nothing else, knowing that a materialized view refreshes daily could prevent adapter from continually advising `persist` to advance compaction, and obliging `persist` to consider the compaction, even though we statically know that nothing could possibly happen.

Reply:
> Have you considered setting the compaction window of objects to the same value as their refresh window? I think that would also help here.

Yeah, sorry this is all I meant. Oops.